examples/session_example.rs

```diff
@@ -0,0 +1,72 @@
+#[macro_use] extern crate nickel;
+extern crate rustc_serialize;
+extern crate time;
+
+use std::io::Write;
+use nickel::*;
+use nickel::status::StatusCode;
+use time::Duration;
+
+#[derive(RustcDecodable, RustcEncodable)]
+struct User {
+    name: String,
+    password:  String,
+}
+
+struct ServerData;
+static SECRET_KEY: &'static cookies::SecretKey = &cookies::SecretKey([0; 32]);
+impl AsRef<cookies::SecretKey> for ServerData {
+    fn as_ref(&self) -> &cookies::SecretKey { SECRET_KEY }
+}
+impl SessionStore for ServerData {
+    type Store = Option<String>;
+
+    fn timeout() -> Duration {
+        Duration::seconds(5)
+    }
+}
+
+
+fn main() {
+    let mut server = Nickel::with_data(ServerData);
+
+    /* Anyone should be able to reach thist route. */
+    server.get("/", middleware! { |mut res|
+        format!("You are logged in as: {:?}\n", res.session())
+    });
+
+    server.post("/login", middleware!{|mut res|
```

Comment:
See https://github.com/nickel-org/nickel-auth/pull/1, it has a test shell script. (but yes we should have a test written in rust!)